src/Psr/CacheItemPool/CacheItemPoolDecorator.php

```diff
@@ -84,7 +83,7 @@ public function __destruct()
     /**
      * {@inheritdoc}
      */
-    public function getItem($key)
+    public function getItem(string $key): CacheItemInterface
```

Comment:
Because this class is not `final`, the change in interface also implicitly becomes a BC break for people not depending on `psr/cache`.

@boesing that means new major version here too :S

Reply:
I think a major version is not needed at this point. The changes to the class `CacheItemPoolDecorator` can be undone, then a minor version will also work.

Reply:
I think a major version is not needed at this point. The changes to the class `CacheItemPoolDecorator` can be undone, then a minor version will also work.

Reply:
I think a major version is not needed at this point. The changes to the class `CacheItemPoolDecorator` can be undone, then a minor version will also work.

Reply:
I think a major version is not needed at this point. The changes to the class `CacheItemPoolDecorator` can be undone, then a minor version will also work.

Reply:
Tend to agree: new major can happen later, if we want :+1:

Reply:
Removed typed arguments and added back psr/cache v1 support.